### PR TITLE
[New Rule] Sublime Plugin or Application Script Modification

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/760

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/persistence_modification_sublime_app_plugin_or_script.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+maturity = "production"
+updated_date = "2020/12/23"
+
+[rule]
+author = ["Elastic"]
+description = """
+Adversaries may create or modify the Sublime application Plugins or Scripts to execute a malicious payload every time
+the Sublime application is started as part of their persistence mechanism.
+"""
+false_positives = ["Legit Sublime Plugins or Application Updates"]
```
Comment (bm11100) [inline rules/macos/persistence_modification_sublime_app_plugin_or_script.toml:12]:
```suggestion
false_positives = ["Legitimate Sublime plugins or application updates"]
```

File: rules/macos/persistence_modification_sublime_app_plugin_or_script.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+[metadata]
+creation_date = "2020/12/23"
+maturity = "production"
+updated_date = "2020/12/23"
+
+[rule]
+author = ["Elastic"]
+description = """
+Adversaries may create or modify the Sublime application Plugins or Scripts to execute a malicious payload every time
+the Sublime application is started as part of their persistence mechanism.
```
Comment (bm11100) [inline rules/macos/persistence_modification_sublime_app_plugin_or_script.toml:10]:
```suggestion
Adversaries may create or modify the Sublime application plugins or scripts to execute a malicious payload each time
the Sublime application is started.
```

Comment (bm11100) [reply rules/macos/persistence_modification_sublime_app_plugin_or_script.toml:10]: re lint

File: rules/macos/persistence_modification_sublime_app_plugin_or_script.toml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+query = '''
+file where event.type in ("change", "creation") and file.extension == "py" and
+ wildcard(file.path, "/Users/*/Library/Application Support/Sublime Text*/Packages/*.py", "/Applications/Sublime Text.app/Contents/MacOS/sublime.py") and
+ not wildcard(process.executable, 
+                 "/Applications/Sublime Text*.app/Contents/MacOS/Sublime Text*", 
+                 "/usr/local/Cellar/git/*/bin/git", 
+                 "/usr/libexec/xpcproxy", 
+                 "/System/Library/PrivateFrameworks/DesktopServicesPriv.framework/Versions/A/Resources/DesktopServicesHelper", 
+                 "/Applications/Sublime Text.app/Contents/MacOS/plugin_host")
```
Comment (bm11100) [inline rules/macos/persistence_modification_sublime_app_plugin_or_script.toml:33]:
```suggestion
file where event.type in ("change", "creation") and file.extension : "py" and
  file.path : 
    (
      "/Users/*/Library/Application Support/Sublime Text*/Packages/*.py", 
      "/Applications/Sublime Text.app/Contents/MacOS/sublime.py"
    ) and
  not process.executable : 
    (
      "/Applications/Sublime Text*.app/Contents/MacOS/Sublime Text*", 
      "/usr/local/Cellar/git/*/bin/git", 
      "/usr/libexec/xpcproxy", 
      "/System/Library/PrivateFrameworks/DesktopServicesPriv.framework/Versions/A/Resources/DesktopServicesHelper", 
      "/Applications/Sublime Text.app/Contents/MacOS/plugin_host"
    )
```

Comment (bm11100) [reply rules/macos/persistence_modification_sublime_app_plugin_or_script.toml:33]: eliminated wildcard functions

**Review by bm11100 — APPROVED** [review]

**Review by threat-punter — APPROVED** [review]

**Pull request merged**